lightning/src/ln/peers/handshake/mod.rs

```diff
@@ -2,85 +2,53 @@
 //! Handshake states can be advanced automatically, or by manually calling the appropriate step.
 //! Once complete, returns an instance of Conduit.
 
-use bitcoin::secp256k1;
-
-use bitcoin::hashes::{Hash, HashEngine};
-use bitcoin::hashes::sha256::Hash as Sha256;
 use bitcoin::secp256k1::{PublicKey, SecretKey};
 
-use ln::peers::{chacha, hkdf};
-use ln::peers::conduit::{Conduit, SymmetricKey};
-use ln::peers::handshake::acts::{ActOne, ActThree, ActTwo, ACT_ONE_LENGTH, ACT_TWO_LENGTH, ACT_THREE_LENGTH, Act};
-use ln::peers::handshake::hash::HandshakeHash;
-use ln::peers::handshake::states::{ActOneExpectation, ActThreeExpectation, ActTwoExpectation, HandshakeState};
+use ln::peers::conduit::Conduit;
+use ln::peers::handshake::acts::Act;
+use ln::peers::handshake::states::{HandshakeState, IHandshakeState};
 
-pub(crate) mod acts;
-mod hash;
+mod acts;
 mod states;
-mod tests;
 
 /// Object for managing handshakes.
 /// Currently requires explicit ephemeral private key specification.
 pub struct PeerHandshake {
 	state: Option<HandshakeState>,
-	private_key: SecretKey,
-	remote_public_key: Option<PublicKey>,
-	ephemeral_private_key: SecretKey,
-
-	read_buffer: Vec<u8>,
+	ready_for_process: bool,
+	last_act: Option<Act>
```

Comment:
`last` is a bit ambivalent. Depending on the usage, it should be either `previous_act` or `latest_act`.

Reply:
I removed this patch from the PR and process_act() will just return a Vec that the handler can forward to the peer. Keeping intermediate bytes around just to return a ref that gets immediately converted to a Vec is just extra confusion for no gain.

I think there is a separate PR for the goal of this type of pattern which was to have the ability to not allocate a vector for these Act messages before they get shipped out since `send_data()` takes a slice. The same comment came up in the original 494 review with the copying of input bytes to the decryptor internal buffer before consuming them resulting in unnecessary allocations.

A more feature-full output buffer that can take in a `Vec<u8>` or a `&[u8]` and properly does the accounting and copying if needed. e.g. If it can fully flush it can just pass through the refs, but if it needs to block it will copy and take ownership of them.

My in-progress PR touches a bit more of that code so I will play around and see if there is anything that might work.